components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/DefaultOAuth2ScopeValidator.java

```diff
@@ -95,6 +95,9 @@ public List<String> validateScope(OAuthAuthzReqMessageContext authzReqMessageCon
         }
         List<String> authorizedScopes = getAuthorizedScopes(requestedScopes, authzReqMessageContext
                         .getAuthorizationReqDTO().getUser(), appId, null, tenantDomain);
+        if (requestedScopes.contains(INTERNAL_LOGIN_SCOPE) && !authorizedScopes.contains(INTERNAL_LOGIN_SCOPE)) {
```

Comment:
Shouldn't we need to add INTERNAL_LOGIN_SCOPE when SYSTEM_SCOPE is in requestedScopes

Reply:
will update